### PR TITLE
Added typed error with tests and implemented them on the client

### PR DESCRIPTION
**Pull request by DirkHeijnen**

### CODE REVIEW

**Comment by DirkHeijnen** [comment]
I opened a new PR for this one because the contributer license agreeement was given some issues on the previous one

File: clients/go/pkg/vexilla/client.go [hunk 1]
```diff
@@ -125,10 +126,11 @@ func (client *Client) GetFlags(groupNameOrId string, fetch func(url string) (*ht
 
 // Sets a fetched flag group within the Client instance.
 func (client *Client) SetFlags(group Group) error {
+
 	groupId := client.groupLookupTable[string(group.GroupId)]
 
 	if groupId == "" {
-		return fmt.Errorf("no group found for: %s", group.GroupId)
+		return NewGroupLookupError("")
```
Comment (cmgriffing) [inline clients/go/pkg/vexilla/client.go]: suggestion: Looks like this should probably be the group.GroupId value rather than an empty string

**Comment by cmgriffing** [comment]
issue: Because we are using Bazel to build and test the Go code, we need to add the new files to the `BUILD.bazel` file in the `internal` folder. Make sure to add the new test file to the test block as well.

**Comment by cmgriffing** [comment]
@DirkHeijnen  THANKS! This is a solid looking PR. I just noticed a couple things to fix whenever you have time.

File: clients/go/pkg/vexilla/client.go (hunk 1 above)
Comment (DirkHeijnen) [reply clients/go/pkg/vexilla/client.go]: Keen eye, added the group.GroupId here!

**Comment by DirkHeijnen** [comment]
No problem, I fixed the mistake with empty string and added the files to bazel!

**Comment by cmgriffing** [comment]
Really great work. Thanks for the contribution.

**Pull request merged**